### PR TITLE
497 need to be able to rename datasets and models from the UI

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Added the possibility to rename dataset and models for admin and project owner

## Related Issue

Migration of component (https://github.com/Giskard-AI/giskard/issues/432):
- Models.vue
- Datasets.vue

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: backend/src/main/java/ai/giskard/service/DatasetService.java [hunk 1]
```diff
@@ -122,4 +124,17 @@ public List<FeatureMetadataDTO> getFeaturesWithDistinctValues(Long datasetId) {
         }).toList();
     }
 
+    @Transactional
+    public Dataset renameDataset(long datasetId, String name) {
+        Dataset dataset = datasetRepository.findById(datasetId)
+            .orElseThrow(() -> new EntityNotFoundException(Entity.DATASET, datasetId));
+
+        if (!SecurityUtils.isCurrentUserAdmin() && !SecurityUtils.isCurrentUserInside(dataset.getProject().getOwner())) {
```
Comment (andreybavt) [inline backend/src/main/java/ai/giskard/service/DatasetService.java:132]: this check is essentially the same as `canWriteProject`, let's use the same access validation as for dataset creation, e.g. `ai/giskard/web/rest/controllers/UploadController.java:56`

File: backend/src/main/java/ai/giskard/web/rest/controllers/ModelController.java [hunk 2]
```diff
@@ -113,4 +119,20 @@ public PredictionDTO predict(@PathVariable @NotNull Long modelId, @RequestBody @
...
         return usageService.prepareDeleteModel(modelId);
     }
+
+    @PatchMapping("models/{modelId}/name/{name}")
+    @Transactional
+    public ModelDTO renameModel(@PathVariable long modelId, @PathVariable @Valid @NotBlank String name) {
+        ProjectModel model = modelRepository.findById(modelId)
+            .orElseThrow(() -> new EntityNotFoundException(Entity.PROJECT_MODEL, modelId));
+
+        if (!SecurityUtils.isCurrentUserAdmin() && !SecurityUtils.isCurrentUserInside(model.getProject().getOwner())) {
```
Comment (andreybavt) [inline backend/src/main/java/ai/giskard/web/rest/controllers/ModelController.java:129]: same comment as for datasets

File: frontend/src/views/main/project/Datasets.vue [hunk 3]
```diff
@@ -71,94 +110,97 @@
...
-      }
-    }
+function editDataset(dataset: FileDTO) {
+  cancelEditingDataset();
+  editedDataset.value = Object.assign({}, dataset);
+}
+
+async function saveDataset() {
+  if (editedDataset.value) {
+    mixpanel.track('Update dataset name', {id: editedDataset.value.id});
```
Comment (andreybavt) [inline frontend/src/views/main/project/Datasets.vue:193]: good job adding tracking 👍

**Comment by andreybavt** [comment]
Since the name is the only property we allow to edit, I'd place the edit/save buttons next to it, otherwise, there's a little back and forth required to enable editing and then save.

image

**Review by andreybavt — CHANGES_REQUESTED** [review]

File: frontend/src/views/main/project/Datasets.vue [hunk 4]
```diff
@@ -71,94 +110,97 @@
...
 
-@Component
-export default class Datasets extends Vue {
-  @Prop({type: Number, required: true}) projectId!: number;
-  @Prop({type: Boolean, default: false}) isProjectOwnerOrAdmin!: boolean;
+const files = ref<FileDTO[]>([]);
+const lastVisitedFileId = ref<number | null>(null);
+const filePreviewHeader = ref<{ text: string, value: string, sortable: boolean }[]>([]);
+const filePreviewData = ref<any[]>([]);
+const editedDataset = ref<FileDTO | null>(null);
```
Comment (andreybavt) [inline frontend/src/views/main/project/Datasets.vue:135]: the type defined should rather be `DatasetDTO`, same for editedDataset

**Comment by kevinmessiaen** [comment]
> Member

I've updated the UI to have the button next to the edited field

**Review by andreybavt — APPROVED** [review]

**Pull request merged**